pkg/controller/machine.go

```diff
@@ -410,37 +410,41 @@ func (c *Controller) deleteMachineAndProviderInstance(prov cloud.Provider, provi
 }
 
 func (c *Controller) createInstanceForMachine(prov cloud.Provider, providerInstance instance.Instance, machine *machinev1alpha1.Machine, userdataProvider userdata.Provider, providerConfig *providerconfig.Config) error {
+	// case 1: validate the machine spec before getting the instance from cloud provider.
+	// even though this is a little bit premature and inefficient, it helps us detect invalid specification
```

Comment:
I don't even think its a big difference in efficiency , because `createInstanceForMachine` is only called when `!nodeExistsAndIsReady`